### DISCUSSION TITLE
docs/virtual-entities

### DISCUSSION THREAD

**Discussion by giscus** [mikro-orm/mikro-orm#6144] [Comments]
# docs/virtual-entities

Virtual entities don't represent any database table. Instead, they dynamically resolve to an SQL query (or an aggregation in MongoDB), allowing to map any kind of results onto an entity. Such entities are meant for read purposes, they don't have a primary key and therefore cannot be tracked for changes. In a way they are similar to (currently unsupported) database views, and you can use them to proxy your native views already.

https://mikro-orm.io/docs/virtual-entities

**Comment by mmick66** [comment]
Please add a basic example on how to use Virtual Entities to query and map data. I would like to see how we pass values to modify the queries and how we use repository or entity manager to load data into those classes.

**Reply by B4nan** [reply]
The very same way as with normal entities, simple `em.find` call.

Here are the tests if you want to look at working code. 

https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/virtual-entities/virtual-entities.postgres.test.ts
